src/components/v5/common/ActionSidebar/partials/forms/EditColonyDetailsForm/utils.tsx

```diff
@@ -14,3 +17,38 @@ export const editColonyDetailsDescriptionMetadataGetter: DescriptionMetadataGett
         : ColonyActionType.ColonyEditMotion,
   });
 };
+
+export const getEditColonyDetailsPayload = (
+  colony: Colony,
+  values: EditColonyDetailsFormValues,
+) => {
+  const {
+    colonyName: colonyDisplayName,
+    colonyDescription,
+    avatar,
+    externalLinks,
+    description: annotationMessage,
+    title,
+  } = values;
+  const { image: colonyAvatarImage, thumbnail: colonyThumbnail } = avatar ?? {};
+
+  return {
+    colony,
+    colonyDisplayName,
+    colonyDescription,
+    colonyAvatarImage:
+      typeof colonyAvatarImage === 'string' || colonyAvatarImage === null
+        ? colonyAvatarImage
+        : colony.metadata?.avatar,
+    colonyThumbnail:
+      typeof colonyThumbnail === 'string' || colonyThumbnail === null
+        ? colonyThumbnail
+        : colony.metadata?.thumbnail,
+    colonyExternalLinks: externalLinks.map(({ url, linkType }) => ({
```

Comment:
just a heads up that this PR and [this PR](https://github.com/JoinColony/colonyCDapp/pull/1530) will probably have conflicts